core/domain/skill_domain.py

```diff
@@ -454,10 +546,27 @@ def validate(self):
                 'received %s' % self.skill_contents)
         self.skill_contents.validate()
 
+        if not isinstance(self.rubrics, list):
+            raise utils.ValidationError(
+                'Expected rubrics to be a list, '
+                'received %s' % self.skill_contents)
+
+        difficulties_list = []
+        for rubric in self.rubrics:
+            if not isinstance(rubric, Rubric):
+                raise utils.ValidationError(
+                    'Expected each rubric to be a Rubric '
+                    'object, received %s' % rubric)
+            if rubric.difficulty in difficulties_list:
+                raise utils.ValidationError(
+                    'Duplicate rubric found for: %s' % rubric.difficulty)
+            difficulties_list.append(rubric.difficulty)
+            rubric.validate()
```

Comment:
Should also validate that the difficulties_list has the correct three elements

Reply:
Actually, scratch that -- I think maybe we should go for a more flexible model and allow skills with fewer than three difficulties (or possibly, in the future, with more than three). That's a more natural way to model the domain.

However I think we should require that at least 1 rubric be specified. This means that, when a skill is created in the frontend, at least one rubric element must be specified by the creator.

@vinitamurthi do you think that's reasonable?

Reply:
Not sure if I understood this. Suppose the creator only specifies Easy and Hard difficulty levels, then do we still keep the mapping that Easy->0.3, Medium->0.6, Hard->0.9 ? That may be a little funny right because a question will either have a 0.3 difficulty or 0.9 difficulty?
Also that would require some frontend changes to the question editor (but that shouldn't be too hard, this is just an FYI)

Reply:
OK, yeah, good point. I haven't thought through all the implications carefully and TBH I'm not sure, so perhaps we had better keep it restricted for now. Let's keep exactly 3 rubrics and do validation for that. Thanks @vinitamurthi!

Reply:
Done